### PR TITLE
Remove use of simplejson throughout code base

### LINKED ISSUE #42761: Remove simplejson support for 2.7
##### SUMMARY

Since python2.4 support has been dropped, simplejson support should no longer be needed, and in some cases actually causes problems/inconsistencies where simplejson is preferred over stdlib json. Remove all simplejson references and fallbacks from both core code and module_utils.

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME

basic.py

##### ANSIBLE VERSION

```
2.7.0
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Remove use of simplejson throughout code base. Fixes #42761

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

many

##### ANSIBLE VERSION

```
2.7
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/module_utils/basic.py [hunk 1]
```diff
@@ -108,27 +108,9 @@
...
-    except SyntaxError:
-        print('\n{"msg": "SyntaxError: probably due to installed simplejson being for a different python version", "failed": true}')
-        sys.exit(1)
-    else:
-        sj_version = json.__version__.split('.')
-        if sj_version < ['1', '6']:
-            # Version 1.5 released 2007-01-18 does not have the encoding parameter which we need
-            print('\n{"msg": "Error: Ansible requires the stdlib json or simplejson >= 1.6.  Neither was found!", "failed": true}')
+    print('\n{"msg": "Error: ansible requires the stdlib json and was not found!", "failed": true}')
+    sys.exit(1)
```
Comment (abadger) [inline lib/ansible/module_utils/basic.py:120]: I'm not entirely sure what we want to do with these... The libs which override json are still present on pypi.  But I bet nowadays they  will break more than ansible so we don't have to worry about them.  On the other hand, I would have thought that when I had to add the code that handled that.

Comment (sivel) [reply lib/ansible/module_utils/basic.py:120]: I've added back in the python-json checks.

**Comment by abadger** [comment]
This looks good +1

**Comment by sivel** [comment]
bot_skip

For > 50 files

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**